trieste/models/gpflux/sampler.py

```diff
@@ -0,0 +1,170 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+import tensorflow as tf
+from gpflux.layers import GPLayer, LatentVariableLayer
+from gpflux.models import DeepGP
+from gpflux.sampling.sample import Sample
+
+from ...types import TensorType
+from ...utils import DEFAULTS
+from ..interfaces import (
+    ProbabilisticModel,
+    ReparametrizationSampler,
+    TrajectoryFunction,
+    TrajectorySampler,
+)
+from .interface import GPfluxPredictor
+
+
+def sample_consistent_lv_layer(layer: LatentVariableLayer) -> Sample:
```

Comment:
Currently this sampler and `sample_dgp` are actually duplicated in `trieste/models/gpflux/utils.py` (it seems that I put them there in the first place...), but I think that it may be more suited here, do you have any particular thoughts? @hstojic

Reply:
in the other PR I also renamed the util file to sampler, so good choice :)